File: android/kssettingsview/src/main/kotlin/jp/kamusoft/kssettingsview/ui/DatePickerUIStyle.kt
package jp.kamusoft.kssettingsview.ui

/**
 * [DatePickerCell] の Android 側 UI スタイルを表す列挙型（Android UI 層所属）。
 *
 * Material と Spinner の UI を切り替えるための論理スイッチで、
 * `AiForms.Maui.SettingsView` の `DatePickerCell.IsAndroidSpinnerStyle` に対応する。
 *
 * - [Material]: 既定。カレンダー選択面（[DateCalendarDialog]、android/ADR-0019）を表示
 * - [Spinner]: ボトムシート + 年/月/日の3連ホイール（[DateSelectionSheet]、android/ADR-0009）を
 *   表示（AiForms の `IsAndroidSpinnerStyle = true` 相当）
 *
 * クロスプラットフォーム命名規約：iOS UI 層にも同名の `DatePickerUIStyle` が存在するが、
 * ケースは別物（iOS: `wheels` / `calendar`）。型名は対称、ケースはプラットフォーム固有 UI を
 * 反映する設計。これにより `DatePickerCell.uiStyle` というプロパティ名を両プラットフォームで
 * 共通化できる。
 */
public enum class DatePickerUIStyle {
    /** Material Design スタイル（カレンダー選択面、既定）。 */
    Material,

    /** Spinner スタイル（ボトムシート + 年/月/日の3連ホイール）。 */
    Spinner,
}

File: maui/KsSettingsView.Maui/DatePickerUIStyle.cs
namespace KsSettingsView;

/// <summary>
/// <see cref="DatePickerCell"/> の選択面の形式 (maui/ADR-0013)。
/// </summary>
/// <remarks>
/// 両 platform の形式を 1 つの列挙で表す。未指定 (null) のときは各 platform の既定に従う。
/// </remarks>
public enum DatePickerUIStyle
{
    /// <summary>カレンダー形式。iOS はカレンダー、Android は Material 形式で表示される。</summary>
    Calendar,

    /// <summary>ホイール形式。iOS はホイール、Android は Spinner 形式で表示される。</summary>
    Wheels,
}
